src/main/java/beam/analysis/summary/VehicleTravelTimeAnalysis.java

```diff
@@ -20,19 +30,47 @@ public void processStats(Event event) {
                     Double.parseDouble(eventAttributes.get(PathTraversalEvent.ATTRIBUTE_DEPARTURE_TIME)));
 
             secondsTraveledByVehicleType.merge(vehicleType, hoursTraveled, (d1, d2) -> d1 + d2);
+
+
+            if (vehicleType.contains("Car") || vehicleType.contains("BUS")){
+                countOfVehicle ++;
+                int numOfPassangers = Integer.parseInt(eventAttributes.get(PathTraversalEvent.ATTRIBUTE_NUM_PASS));
+
+                double freeFlowDuration = 0.0;
+                Map<Id<Link>, ? extends  Link> linkslist =  scenario.getNetwork().getLinks();
```

Comment:
This code is dangerous as scenario is not guaranteed to be set. I'm assuming you only adding a constructor overload for backwards compatibility or something, but now this may blow up for those areas which use the old constructor. Should `scenario` be *required*, or should this line become an if check that returns empty if it doesn't exist?

Reply:
Hi @JustinPihony , This class `VehicleTravelTimeAnalysis` is only used in `StatsFactory` and I have add the parameter there. Also we need `scenario` for `freeFlowDuration` and I have added a null check to this.